frontend/src/components/layout/Header/ErrorModal.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { Row, Col, Modal, Button } from 'react-bootstrap';
+import { IGenericNetworkAction, clear } from 'actions/genericActions';
+
+export interface IErrorModalProps {
+  // An array of network action errors.
+  errors: IGenericNetworkAction[];
+  // Whether to show the modal.
+  show?: boolean;
+  // Change the visible state.
+  setShow: (show: boolean) => void;
+}
+
+/**
+ * ErrorModal component that displays an error message and information in a modal.
+ * @param param0 ErrorModal component properties.
+ * @param param0.errors An array of errors.
+ * @param [param0.show] Whether to show the model.
```

Comment:
is something weird going on with your autogenerated comments?

Reply:
I manually added the second two based on jsdoc.  Thinking about it now, I probably don't need to as they will inherit the documentation from the interface automatically.

Reply:
what is with the []?

Reply:
optional parameter